### PR TITLE
support placeholder at apiReturnListType and apiReturnType

### PR DESCRIPTION
**Pull request by matsudamper**
---

### Description

Related to #1167

Added placeholder support for apiReturnListType and apiReturnType.
The following syntax is supported.
```
apiReturnType = "java.util.concurrent.CompletionStage<graphql.execution.DataFetcherResult<{{TYPE}}>>"
```

---

Changes were made to:
- [x] Codegen library - Java
- [x] Codegen library - Kotlin
- [x] Codegen library - Scala
- [ ] Maven plugin
- [ ] Gradle plugin
- [ ] SBT plugin

### CODE REVIEW

File: src/main/java/com/kobylynskyi/graphql/codegen/java/JavaGraphQLTypeMapper.java [hunk 1]
```diff
@@ -65,11 +69,25 @@ public String wrapApiReturnTypeIfRequired(MappingContext mappingContext,
         if (computedTypeName.startsWith(JAVA_UTIL_LIST) &&
                 Utils.isNotBlank(mappingContext.getApiReturnListType())) {
             // in case it is query/mutation, return type is list and apiReturnListType is set
-            return computedTypeName.replace(JAVA_UTIL_LIST, mappingContext.getApiReturnListType());
+            if (mappingContext.getApiReturnListType().contains(MappingConfigConstants.API_RETURN_NAME_PLACEHOLDER)) {
+                Matcher matcher = JAVA_UTIL_LIST_ELEMENT_REGEX.matcher(computedTypeName);
+                matcher.find();
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/java/JavaGraphQLTypeMapper.java:74]: @matsudamper I think it's better if you do `if (matcher.find()) { ... }`

File: src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java [hunk 2]
```diff
@@ -92,11 +96,32 @@ public String wrapApiReturnTypeIfRequired(MappingContext mappingContext,
         if (computedTypeName.startsWith(KOTLIN_UTIL_LIST) &&
                 Utils.isNotBlank(mappingContext.getApiReturnListType())) {
             // in case it is query/mutation, return type is list and apiReturnListType is set
-            return computedTypeName.replace(KOTLIN_UTIL_LIST, mappingContext.getApiReturnListType());
+            if (mappingContext.getApiReturnListType().contains(MappingConfigConstants.API_RETURN_NAME_PLACEHOLDER)) {
+                boolean isNullable = computedTypeName.endsWith(KOTLIN_UTIL_NULLABLE);
+
+                Matcher matcher = KOTLIN_UTIL_LIST_ELEMENT_REGEX.matcher(computedTypeName);
+                matcher.find();
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java:103]: @matsudamper I think it's better if you do `if (matcher.find()) { ... }`

File: src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLTypeMapper.java [hunk 3]
```diff
@@ -72,11 +76,25 @@ public String wrapApiReturnTypeIfRequired(MappingContext mappingContext,
         if (computedTypeName.startsWith(SCALA_UTIL_LIST) &&
                 Utils.isNotBlank(mappingContext.getApiReturnListType())) {
             // in case it is query/mutation, return type is list and apiReturnListType is set
-            return computedTypeName.replace(SCALA_UTIL_LIST, mappingContext.getApiReturnListType());
+            if (mappingContext.getApiReturnListType().contains(MappingConfigConstants.API_RETURN_NAME_PLACEHOLDER)) {
+                Matcher matcher = SCALA_UTIL_LIST_ELEMENT_REGEX.matcher(computedTypeName);
+                matcher.find();
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLTypeMapper.java:81]: @matsudamper I think it's better if you do `if (matcher.find()) { ... }`

**Review by kobylynskyi — CHANGES_REQUESTED** [review]
@matsudamper the approach looks good to me. Can you please fix a minor comment I have in Java/Kotlin/Scala mappers and we should be good to merge this. Thanks

File: src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLTypeMapper.java (hunk 3 above)
Comment (matsudamper) [reply src/main/java/com/kobylynskyi/graphql/codegen/scala/ScalaGraphQLTypeMapper.java]: Implemented.
https://github.com/kobylynskyi/graphql-java-codegen/pull/1200/commits/2542b8f5de5bc5660372d7424c60c83b44e2b0c3

**Review by kobylynskyi — APPROVED** [review]
LGTM! Will include this feature in the next release. Thank you for the contribution!

**Pull request merged**